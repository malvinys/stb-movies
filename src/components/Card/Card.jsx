import React from 'react';
import { useDispatch } from 'react-redux';
import PropTypes from 'prop-types';
import { LazyLoadImage } from 'react-lazy-load-image-component';

import { setModalMovieImage } from '../../models/Movies';

const Card = ({
  id, title, image, rating, releaseDate,
}) => {
  const dispatch = useDispatch();

  const handleImageClick = (imageUrl) => {
    dispatch(setModalMovieImage(imageUrl));
  };

  return (
    <>
      <div className="base-card">
        <div className="base-card-image">
          <LazyLoadImage src={image} alt={title} effect="blur" onClick={() => handleImageClick(image)} />
        </div>
        {rating && <div className="base-card-rating">{rating}</div>}
        <a href={`/${id}`}>
          <div className="base-card-content">
            <h2>{title}</h2>
            <p>{releaseDate}</p>
          </div>
        </a>
      </div>
    </>
  );
};

Card.propTypes = {
  id: PropTypes.string,
  title: PropTypes.string,
  image: PropTypes.string,
  rating: PropTypes.number,
  releaseDate: PropTypes.string,
};

Card.defaultProps = {
  id: null,
  title: null,
  image: null,
  rating: null,
  releaseDate: null,
};

export default Card;
