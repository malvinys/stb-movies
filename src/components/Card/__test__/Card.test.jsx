import * as React from 'react';
import { createRenderer } from 'react-test-renderer/shallow';
import { Provider } from 'react-redux';

import store from '../../../models/_store';

import Card from '../Card';

const shallowRenderer = createRenderer();

describe('<Card />', () => {
  it('should match snapshot', () => {
    shallowRenderer.render(
      <Provider store={store}>
        <Card
          id="testId"
          title="test"
          image="test.jpg"
          rating={8}
          releaseDate="2021"
        />
      </Provider>,
    );
    const renderedOutput = shallowRenderer.getRenderOutput();
    expect(renderedOutput).toMatchSnapshot();
  });
});
