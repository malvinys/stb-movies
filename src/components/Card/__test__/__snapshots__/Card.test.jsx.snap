// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Card /> should match snapshot 1`] = `
<Context.Provider
  value={
    Object {
      "store": Object {
        "@@observable": [Function],
        "dispatch": [Function],
        "getState": [Function],
        "replaceReducer": [Function],
        "subscribe": [Function],
      },
      "subscription": Subscription {
        "handleChangeWrapper": [Function],
        "listeners": Object {
          "notify": [Function],
        },
        "onStateChange": [Function],
        "parentSub": undefined,
        "store": Object {
          "@@observable": [Function],
          "dispatch": [Function],
          "getState": [Function],
          "replaceReducer": [Function],
          "subscribe": [Function],
        },
        "unsubscribe": null,
      },
    }
  }
>
  <Card
    id="testId"
    image="test.jpg"
    rating={8}
    releaseDate="2021"
    title="test"
  />
</Context.Provider>
`;
